src/uct/ib/base/ib_iface.c

```diff
@@ -636,11 +636,17 @@ struct ibv_cq *uct_ib_create_cq(struct ibv_context *context, int cqe,
         cq_attr.flags = IBV_CREATE_CQ_ATTR_IGNORE_OVERRUN;
     }
 
-    cq = ibv_cq_ex_to_cq(ibv_create_cq_ex(context, &cq_attr));
-#else
-    cq = ibv_create_cq(context, cqe, NULL, channel, comp_vector);
+    cq_ex = ibv_create_cq_ex(context, &cq_attr);
+    if (cq_ex) {
+        return ibv_cq_ex_to_cq(cq_ex);
+    } else if (errno != ENOSYS) {
+        return NULL;
+    } else
+        /* if ibv_create_cq_ex returns ENOSYS, fallback to ibv_create_cq */
 #endif
-    return cq;
+    {
+        return ibv_create_cq(context, cqe, NULL, channel, comp_vector);
```

Comment:
minor: why not just leave this line without ```{ }```? as I can see there is return from function if cq_ex is completed successfully or failed due to error

Reply:
like
```
    if (cq_ex) {
        return ibv_cq_ex_to_cq(cq_ex);
    } else if (errno != ENOSYS) {
        return NULL;
    }
    /* if ibv_create_cq_ex returns ENOSYS, fallback to ibv_create_cq */
#endif
    return ibv_create_cq(context, cqe, NULL, channel, comp_vector);
```